daemon/containerd/image_inspect.go

```diff
@@ -70,7 +70,11 @@ func (i *ImageService) ImageInspect(ctx context.Context, refOrID string, _ backe
 
 		repoTags = append(repoTags, reference.FamiliarString(name))
 		if _, ok := name.(reference.Digested); ok {
-			repoDigests[reference.FamiliarString(name)] = struct{}{}
+			d := reference.FamiliarString(name)
+			if _, ok := uniqueRepoDigests[d]; !ok {
+				repoDigests = append(repoDigests, d)
+				uniqueRepoDigests[d] = struct{}{}
+			}
```

Comment:
As these slices are generally not that large, even wonder if using `slices.Contains(d)` would do the trick, i.e.,


```go
if d := reference.FamiliarString(name); !slices.Contains(repoDigests, d) {
	repoDigests = append(repoDigests, d)
}
```

Reply:
Perhaps the `slices.Contains()` is actually the most readable approach, and `repoDigests` would usually only contain few entries (only if same digest, but multiple repositories)